ethcore/evm/src/interpreter/gasometer.rs

```diff
@@ -232,8 +232,20 @@ impl<Gas: evm::CostType> Gasometer<Gas> {
 				Request::GasMemProvide(gas, mem, Some(requested))
 			},
 			instructions::CREATE | instructions::CREATE2 => {
-				let gas = Gas::from(schedule.create_gas);
-				let mem = mem_needed(stack.peek(1), stack.peek(2))?;
+				let start = stack.peek(1);
+				let len = stack.peek(2);
+
+				let gas = match instruction {
```

Comment:
might be better to just split `instructions::CREATE` and `instructions::CREATE2` in the upper match instead of matching twice.